### PR TITLE
test(robot): migrate test_setting_concurrent_rebuild_limit

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/8259

#### What this PR does / why we need it:

migrate test_setting_concurrent_rebuild_limit

### CODE REVIEW

File: e2e/tests/regression/test_settings.robot [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+*** Variables ***
+${LOOP_COUNT}    1
+${RETRY_COUNT}    300
+${RETRY_INTERVAL}    1
+
+*** Test Cases ***
+Test Setting Concurrent Rebuild Limit
+    [Documentation]    Test if setting Concurrent Replica Rebuild Per Node Limit works correctly.
+    When Set setting concurrent-replica-rebuild-per-node-limit to 1
```
Comment (c3y1huang) [inline e2e/tests/regression/test_settings.robot:21]: Please follow the BDD style, adopting the Given, When, Then structure.

https://github.com/longhorn/longhorn/issues/8375

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: e2e/tests/regression/test_settings.robot (hunk 1 above)
Comment (yangchiu) [reply e2e/tests/regression/test_settings.robot]: Refined. Thank you!

File: e2e/libs/volume/rest.py [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
+import asyncio
 from volume.base import Base
 from volume.constant import DEV_PATH
-from volume.constant import RETRY_COUNTS
-from volume.constant import RETRY_INTERVAL
 from volume.constant import VOLUME_FRONTEND_BLOCKDEV
 from volume.constant import VOLUME_FRONTEND_ISCSI
-
+from utility.constant import LONGHORN_NAMESPACE
+from utility.utility import get_retry_count_and_interval
```
Comment (roger-ryao) [inline e2e/libs/volume/rest.py:9]: Hi @yangchiu 
    How can we modify the `retry_count` and `interval` for debugging purposes? 
    Can we execute a command with `-v RETRY_COUNT:600` or is there a file where we can modify the value?
     https://github.com/longhorn/longhorn-tests/blob/d6e8928259d3485a1c6cbe1e5ca369ac6697bb2a/e2e/libs/utility/utility.py#L47-L50

Comment (roger-ryao) [reply e2e/libs/volume/rest.py:9]: I found the answer by referring to https://github.com/longhorn/longhorn-tests/blob/d6e8928259d3485a1c6cbe1e5ca369ac6697bb2a/e2e/tests/regression/test_settings.robot#L15

File: e2e/libs/keywords/volume_keywords.py [hunk 3]
```diff
@@ -159,6 +161,62 @@ def wait_for_replica_rebuilding_to_complete(self, volume_name):
...
+        first_replica_rebuilding, not_start_replica_rebuilding = await find_first_replica_rebuilding()
+        logging(f"Observed {first_replica_rebuilding} started replica rebuilding first")
+
+        while self.volume.is_replica_rebuilding_in_progress(first_replica_rebuilding, node_id):
+            logging(f"Checking volume {not_start_replica_rebuilding} replica rebuilding won't start \
+                if volume {first_replica_rebuilding} replica rebuilding is still in progress")
+            assert not self.volume.is_replica_rebuilding_in_progress(not_start_replica_rebuilding, node_id)
+            time.sleep(1)
+
+    async def both_replica_rebuildings_will_start_at_the_same_time_on_node(self, volume_name_0, volume_name_1, replica_locality):
```
Comment (roger-ryao) [inline e2e/libs/keywords/volume_keywords.py:191]: Hi @yangchiu 
       Do you think it's necessary to add a `timeout` to ensure that the `both_replica_rebuildings_will_start_at_the_same_time_on_node` &  `only_one_replica_rebuilding_will_start_at_a_time_on_node` doesn't get stuck indefinitely?

Comment (yangchiu) [reply e2e/libs/keywords/volume_keywords.py:191]: Each underlay function has its own timeout, like the retry_count in `wait_for_replica_rebuilding_start`. Does it make sense to you?

Comment (roger-ryao) [reply e2e/libs/keywords/volume_keywords.py:191]: > Each underlay function has its own timeout, like the retry_count in `wait_for_replica_rebuilding_start`. Does it make sense to you?

LGTM

**Review by roger-ryao — APPROVED** [review]
LGTM

**Pull request merged**